engine/rotationmanager/advance.go

```diff
@@ -35,6 +38,11 @@ func calcAdvance(t time.Time, rot *rotation.Rotation, state rotation.State, part
 		return nil
 	}
 
+	warningThreshold := time.Minute * -15
+	if !newStart.After(t.Add(warningThreshold)) {
```

Comment:
Use `value * unit` for readability, also no need for a variable, it is only used once, and doesn't change
```suggestion
	if !newStart.After(t.Add(-15 * time.Minute)) {
```